react/StarIcon/StarIcon.less

```diff
@@ -9,3 +19,59 @@
   50% { transform: scale(1.1); }
   100% { transform: scale(1); }
 }
+
+._large {
+  .svg {
+    position: relative;
+    top: 0.05em;
+  }
+
+  &.hero {
+    .iconHeight(@hero-type-row-span, @hero-type-row-span-mobile);
```

Comment:
@michaeltaranto just a heads up, we had to explicitly set the height of the icon container so that it follows the grid, rather than relying on child nodes. Ideally, this would happen in `Icon.less`, but wanted to limit the blast radius of this PR—we'll revisit this when we revisit icons as a whole.